File: Daily Test/Python/Sort Submatrices - Alphabets 7.2.2026.py
r, c = map(int, input().split())
mat = [input().split() for _ in range(r)]
k = int(input())

for i in range(0, r, k):
    for j in range(0, c, k):
        res = []
        for l in range(i, i + k):
            for m in range(j, j + k):
                res.append(mat[l][m])
        res.sort()
        u = 0
        for o in range(i, i + k):
            for p in range(j, j + k):
                mat[o][p] = res[u]
                u += 1

for row in mat:
    print(*row)

File: Daily Test/Java/Sort Submatrices - Alphabets 7.2.2026.java
import java.util.*;

public class Hello {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int r = sc.nextInt();
        int c = sc.nextInt();

        String[][] mat = new String[r][c];
        for (int i = 0; i < r; i++)
            for (int j = 0; j < c; j++)
                mat[i][j] = sc.next();

        int k = sc.nextInt();

        for (int i = 0; i < r; i += k) {
            for (int j = 0; j < c; j += k) {
                ArrayList<String> res = new ArrayList<>();

                for (int x = i; x < i + k; x++)
                    for (int y = j; y < j + k; y++)
                        res.add(mat[x][y]);

                Collections.sort(res);

                int idx = 0;
                for (int x = i; x < i + k; x++)
                    for (int y = j; y < j + k; y++)
                        mat[x][y] = res.get(idx++);
            }
        }

        for (String[] row : mat) {
            for (String s : row)
                System.out.print(s + " ");
            System.out.println();
        }
    }
}